examples/pytorch/ARGO/README.md

```diff
@@ -0,0 +1,196 @@
+# ARGO: An Auto-Tuning Runtime System for Scalable GNN Training on Multi-Core Processor
+
+## Overview
+
+Graph Neural Network (GNN) training suffer from low scalability on multi-core processors. 
+ARGO is a runtime system that offers scalable performance. 
+Shown in the figure below, without ARGO, there is no performance improvement after applying more than 16 cores. 
+However, with ARGO enabled, we are able to scale over 64 cores, allowing ARGO to speedup GNN training (in terms of epoch time) by up to 4.30x.
+![ARGO](https://github.com/jasonlin316/dgl/blob/argo/examples/pytorch/ARGO/argo_scale.png)
+
+
+
+This README includes how to:
+1. [Set up the environment](#1-setting-up-the-environment)
+2. [Run the example code](#2-running-the-example-GNN-program)
+3. [Modify your own GNN program to enable ARGO.](#3-enabling-ARGO-on-your-own-GNN-program)
+
+## 1. Setting up the environment
+
+1. Clone the repository:
+
+   ```shell
+   git clone https://github.com/jasonlin316/ARGO.git
+   ```
+
+2. Download Anaconda and install
+   ```shell
+   wget https://repo.anaconda.com/archive/Anaconda3-2023.03-Linux-x86_64.sh
+   bash Anaconda3-2023.03-Linux-x86_64.sh
+   ```
+
+3. Create a virtual environment:
+
+   ```shell
+   conda create -n py38 python=3.8.1
+   ```
+
+4. Active the virtual environment:
+
+   ```shell
+   conda activate py38
+   ```
+
+5. Install required packages:
+
+   ```shell
+   conda install pytorch torchvision torchaudio cpuonly -c pytorch
+   conda install -c dglteam dgl
+   conda install -c conda-forge ogb
+   conda install -c conda-forge scikit-optimize
+   ```
+>  Note: there exist a bug in the older version (before v0.9.0) of the Scikit-Optimization library.  
+To fix the bug, find the "transformer.py" which should be located in  
+   ```~/anaconda3/envs/py38/lib/python3.8/site-packages/skopt/space/transformers.py```  
+Once open the file, replace all ```np.int_``` with ```int```.
```

Comment:
I think it should be fine. The latest scikit-optimize fixed this bug already. The note was there just in case anyone installed an older version by any chance.

Reply:
Unfortunately the bug still exist as now "np.int" is also not allowed by numpy ((https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated) .
Numpy  1.26.3 return following error:
```
>>> import numpy as np
>>> np.int
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/miniconda3/envs/argo/lib/python3.10/site-packages/numpy/__init__.py", line 324, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'inf'?
>>> 
```

Reply:
Hi @anko-intel, I monkey-patched the problematic functions. Basically, we define our own skopt functions that do not use np.int, and force ARGO to use our version. 
Can you try testing it again? Thanks.

Reply:
Hi @jasonlin316, thanks for the patch, it works well.